daemon/daemon.go

```diff
@@ -970,7 +970,7 @@ func NewDaemon(dp datapath.Datapath) (*Daemon, *endpointRestoreState, error) {
 	// Kubernetes apiserver and serving the API to ensure service IDs are
```

Comment:
FYI I have a separate PR out for this - 
https://github.com/cilium/cilium/pull/7384

I put it here because running the Daemon unit tests was a huge pain in this PR, so once 7384 is merged I can rebase on top of master and this commit will go away.